Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -514,88 +514,14 @@ dtStatus dtNavMeshQuery::findRandomPointAroundCircle(dtPolyRef startRef, const f
 dtStatus dtNavMeshQuery::closestPointOnPoly(dtPolyRef ref, const float* pos, float* closest, bool* posOverPoly) const
```

Comment:
Should update `dtNavMeshQuery::getPolyHeight` too.

Reply:
Right. Not sure what the best way is -- if internal points can fail all the triangle checks then we need an additional poly check to figure out whether the point is inside the poly. At that point it is basically the same as `closestPointOnPoly`.

Reply:
This code seems is a bit weird: https://github.com/recastnavigation/recastnavigation/blob/3a619d773deb7e3a15ee215217c825995fe71312/Detour/Source/DetourNavMeshQuery.cpp#L680-L689
For instance, if you call it with a point that is far away, it returns approximately the mid point. Shouldn't this just be using `dtDistancePtSegSqr2D` too?

Reply:
Yes, it should.

Reply:
On the other hand I am not sure how useful this behavior is (returning true for off-mesh connections). I have actually broken it right now. Oh well, I will restore it before merging this.